DDDEastAnglia/Areas/Admin/Controllers/SponsorController.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        private readonly ISponsorRepository sponsorRepository;
+        private readonly ISponsorSorter sponsorSorter;
+
+        public SponsorController(ISponsorRepository sponsorRepository, ISponsorSorter sponsorSorter)
+        {
+            if (sponsorRepository == null)
+            {
+                throw new ArgumentNullException("sponsorRepository");
+            }
+
+            if (sponsorRepository == null)
+            {
+                throw new ArgumentNullException("sponsorRepository");
+            }
+
+            this.sponsorRepository = sponsorRepository;
+            this.sponsorSorter = sponsorSorter;
+        }
+
+        public ActionResult Index()
+        {
+            var sponsors = sponsorRepository.GetAllSponsors();
+            var sortedSponsors = sponsorSorter.Sort(sponsors)
+                                              .Select(CreateSponsorModel)
+                                              .ToList();
+            return View(sortedSponsors);
+        }
+
+        public ActionResult Create()
+        {
+            return View(new SponsorModel());
+        }
+
+        [HttpPost]
+        public ActionResult Create(SponsorModel sponsor)
+        {
+            sponsorRepository.AddSponsor(CreateSponsor(sponsor));
+            return RedirectToAction("Index");
+        }
+
+        public ActionResult Delete(int id)
+        {
+            var sponsor = sponsorRepository.GetSponsor(id);
+            return sponsor == null ? (ActionResult)HttpNotFound() : View(sponsor);
+        }
+
+        [HttpPost, ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public ActionResult DeleteConfirmed(int id)
+        {
+            sponsorRepository.DeleteSponsor(id);
+            return RedirectToAction("Index");
+        }
+
+        public ActionResult Logo(int sponsorId)
+        {
+            var sponsor = sponsorRepository.GetSponsor(sponsorId);
+            return File(sponsor.Logo, "image/png");
+        }
+
+        private SponsorModel CreateSponsorModel(Sponsor sponsor)
+        {
+            return new SponsorModel
+            {
+                SponsorId = sponsor.SponsorId,
+                Name = sponsor.Name,
+                Url = sponsor.Url,
+                SponsorshipAmount = sponsor.SponsorshipAmount,
+                PaymentDate = sponsor.PaymentDate,
+                ShowPublicly = sponsor.ShowPublicly
+            };
+        }
+
+        private Sponsor CreateSponsor(SponsorModel sponsorModel)
+        {
+            return new Sponsor
+            {
+                Name = sponsorModel.Name,
+                Url = sponsorModel.Url,
+                SponsorshipAmount = sponsorModel.SponsorshipAmount,
+                Logo = GetLogo(sponsorModel.Logo),
+                PaymentDate = sponsorModel.PaymentDate,
+                ShowPublicly = sponsorModel.ShowPublicly
+            };
+        }
+
+        private byte[] GetLogo(HttpPostedFileBase file)
```

Comment:
This isn't a `Controller` concern, it's a concern of either the Sponsor service, or an Image infrastructure service.

([Jimmy Bogard's blog post](https://lostechies.com/jimmybogard/2008/08/21/services-in-domain-driven-design/) gives a good overview of the different types of service: application, domain and infrastructure.)